src/main/java/javaslang/collection/Iterator.java

```diff
@@ -166,6 +167,16 @@ public T next() {
         };
     }
 
+    @Override
+    default Iterator<T> iterator() {
+        return this;
```

Comment:
Oh, I've overseen it. Then iterator() stays with us and the ofAll() methods pass away. (Note: we still need ofAl(T...))